crates/sequencer/src/sequencer.rs

```diff
@@ -1032,12 +1032,12 @@ where
             .set_last_sequencer_commitment_l2_height(BatchNumber(db_commitment.l2_end_height))?;
 
         let batches = self.ledger_db.get_soft_batch_range(
-            &(BatchNumber(db_commitment.l2_start_height)
-                ..BatchNumber(db_commitment.l2_start_height + 1)),
+            &(BatchNumber(db_commitment.l2_end_height)
+                ..BatchNumber(db_commitment.l2_end_height + 1)),
         )?;
-        let start_batch = &batches[0];
+        let l2_end_batch = &batches[0];
```

Comment:
You're still using the first one here.

Reply:
`batches` variable only contains one batch. The range statement I used to query the batches is:
```
&(BatchNumber(db_commitment.l2_end_height)
                ..BatchNumber(db_commitment.l2_end_height + 1))
```

So it is actually only getting one batch, which is the l2 end batch.